### PR TITLE
Disable/re-enable rules from within source code comments. Fixes #4.

### LINKED ISSUE #4: Support disabling/enabling rules from source code comments
For example:

```swift
// swiftlint:disable_rule:line_length
... some long line
// swiftlint:enable_rule:line_length
```

### PR DESCRIPTION
**Pull request by jpsim**
First draft addressing #4. Still need to write some tests, but seems to be working from what I can tell from manually testing. /cc @keith @segiddins

### CODE REVIEW

File: Source/SwiftLintFramework/File+SwiftLint.swift [hunk 1]
```diff
@@ -9,9 +9,51 @@
...
+    case Enable = "enable"
+    case Disable = "disable"
+}
+
+public typealias Command = (CommandAction, String, Int)
 
 extension File {
+    public func regions() -> [Region] {
+        let nsStringContents = contents as NSString
+        let commands = matchPattern("swiftlint:(enable|disable):force_cast")
```
Comment (segiddins) [inline Source/SwiftLintFramework/File+SwiftLint.swift]: shouldn't `force_cast` be `.+` and then captured to determine which rule it is?

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,11 @@
   names to start with an underscore.
   [JP Simard](https://github.com/jpsim)
 
+* Disable and re-enable rules from within source code comments using
+  `// swiftlint:disable:$IDENTIFIER` and `// swiftlint:enable:$IDENTIFIER`.
```
Comment (segiddins) [inline CHANGELOG.md]: i'd prefer `// swiftlint:disable force_unwrap`

File: Source/SwiftLintFramework/File+SwiftLint.swift (hunk 1 above)
Comment (jpsim) [inline Source/SwiftLintFramework/File+SwiftLint.swift]: the rule capturing was already being done by `NSScanner`, but I had accidentally hardcoded `force_cast` in here as part of my testing.

File: CHANGELOG.md (hunk 2 above)
Comment (jpsim) [inline CHANGELOG.md]: that *would* make it more similar to rubocop, and I have no main reason to keep that separator a colon... so ok!

**Comment by jpsim** [comment]
Good enough! :shipit:

**Pull request merged**